src/Microsoft.PowerApps.TestEngine/TestEngine.cs

```diff
@@ -123,6 +123,18 @@ public async Task<string> RunTestAsync(FileInfo testConfigFile, string environme
             {
                 if (TestLoggerProvider.TestLoggers.ContainsKey(testRunId))
                 {
+                    if (!string.IsNullOrEmpty(testRunDirectory))
+                    {
+                        if (!Directory.Exists(testRunDirectory))
+                        {
+                            _fileSystem.CreateDirectory(testRunDirectory);
```

Comment:
I would try to keep File/Directory operations as less as possible inside a finally block since these events again tend to throw exceptions. I would rather initialize these folders/files in the try block itself and just operate on them here (that way it is guaranteed that the folder exists even before we get into the finally block). Basically having throwable expressions inside a finally block is additional scope for failure and a maintenance overhead.